test/oban/live_dashboard_test.exs

```diff
@@ -14,23 +14,75 @@ defmodule Oban.LiveDashboardTest do
   test "shows jobs with limit" do
     for _ <- 1..110, do: job_fixture()
     {:ok, live, rendered} = live(build_conn(), "/dashboard/oban")
-    assert rendered |> :binary.matches("<td class=\"oban-jobs-id\"") |> length() <= 100
+
+    assert rendered |> :binary.matches("<td class=\"oban-jobs-all-worker\"") |> length() ==
+             20
 
     rendered = render_patch(live, "/dashboard/oban?limit=100")
-    assert rendered |> :binary.matches("<td class=\"oban-jobs-id\"") |> length() == 100
+
+    assert rendered |> :binary.matches("<td class=\"oban-jobs-all-worker\"") |> length() ==
+             100
+  end
+
+  test "switch between states" do
+    Oban.LiveDashboardTest.Repo.delete_all(Oban.Job)
+    _executing_job = job_fixture(%{"foo" => "executing"}, "executing")
+    _completed_job = job_fixture(%{"foo" => "completed"}, "completed")
+
+    conn = build_conn()
+    {:ok, live, rendered} = live(conn, "/dashboard/oban")
+
+    assert rendered |> :binary.matches("<td class=\"oban-jobs-all-worker\"") |> length() == 2
+
+    {:ok, live, rendered} =
+      live
+      |> element("a", "executing - (1)")
+      |> render_click()
+      |> follow_redirect(conn)
+
+    assert rendered
+           |> :binary.matches("<td class=\"oban-jobs-executing-worker\"")
+           |> length() == 1
+
+    {:ok, live, rendered} =
+      live
+      |> element("a", "completed - (1)")
+      |> render_click()
+      |> follow_redirect(conn)
+
+    assert rendered
+           |> :binary.matches("<td class=\"oban-jobs-completed-worker\"")
+           |> length() == 1
+
+    {:ok, _live, rendered} =
+      live
+      |> element("a", "scheduled - (0)")
+      |> render_click()
+      |> follow_redirect(conn)
+
+    assert rendered
+           |> :binary.matches("<td class=\"oban-jobs-scheduled-worker\"")
+           |> length() == 0
   end
 
   test "shows job info modal" do
     job = job_fixture(%{something: "foobar"})
-    {:ok, live, rendered} = live(build_conn(), "/dashboard/oban?params[job]=#{job.id}")
+    {:ok, live, _rendered} = live(build_conn(), "/dashboard/oban?params[job]=#{job.id}")
     rendered = render(live)
     assert rendered =~ "modal-content"
     assert rendered =~ "%{&quot;something&quot; =&gt; &quot;foobar&quot;}"
     refute live |> element("#modal-close") |> render_click() =~ "modal"
   end
 
-  defp job_fixture(args \\ %{}) do
-    {:ok, job} = Oban.Job.new(args, worker: "FakeWorker") |> Oban.insert()
+  defp job_fixture(args \\ %{}, state \\ "executing") do
+    {:ok, job} =
+      Oban.Job.new(args,
+        worker: "FakeWorker",
+        state: state,
+        attempted_at: DateTime.utc_now()
+      )
+      |> Oban.insert()
```

Comment:
`attempted_at` is not present for all job states. Instead, it would be more appropriate to set those fields when required.
```suggestion
  defp job_fixture(args \\ %{}, opts \\ []) do
    opts  = Keyword.put(opts, :worker, "FakeWorker")
    {:ok, job} = Oban.Job.new(args, opts) |> Oban.insert()
```